qute.jdt/com.redhat.qute.jdt/src/main/java/com/redhat/qute/jdt/QuteSupportForTemplate.java

```diff
@@ -566,5 +562,122 @@ public WorkspaceEdit generateMissingJavaMember(GenerateMissingJavaMemberParams p
+	public String getJavadoc(QuteJavadocParams params, IJDTUtils utils, IProgressMonitor monitor, Set<String> visited) throws JavaModelException {
+		if (visited.contains(params.getClassName())) {
+			return null;
+		}
+		visited.add(params.getClassName());
+		if (monitor.isCanceled()) {
+			throw new OperationCanceledException();
+		}
+		
+		IType type = getTypeFromParams(params.getProjectUri(), params.getClassName(), monitor);
+		if (type == null) {
+			return null;
+		}
+
+		ITypeResolver typeResolver = createTypeResolver(type);
+		
+		// 1) Check the fields for the member
+
+		// Standard fields
+		IField[] fields = type.getFields();
+		for (IField field : fields) {
+			if (isValidField(field, type)
+					&& params.getMemberName().equals(field.getElementName())
+					&& params.getSignature().equals(typeResolver.resolveFieldSignature(field))) {
+				String javadoc = utils.getJavadoc(field, params.getDocumentFormat()); // TODO:
+				if (javadoc != null) {
+					return javadoc;
+				}
+			}
+		}
+
+		// Record fields
+		if (type.isRecord()) {
+			for (IField field : type.getRecordComponents()) {
+				// All record components are valid
+				if (params.getMemberName().equals(field.getElementName())
+						&& params.getSignature().equals(typeResolver.resolveFieldSignature(field))) {
+					String javadoc = utils.getJavadoc(field, params.getDocumentFormat());
+					if (javadoc != null) {
+						return javadoc;
+					}
+				}
+			}
+		}
+
+		// 2) Check the methods for the member
+		IMethod[] methods = type.getMethods();
+		for (IMethod method : methods) {
+			if (isValidMethod(method, type)) {
+				try {
+					InvalidMethodReason invalid = getValidMethodForQute(method, params.getClassName());
+					if (invalid == null && params.getSignature().equals(typeResolver.resolveMethodSignature(method))) {
+						String javadoc = utils.getJavadoc(method, params.getDocumentFormat());
+						if (javadoc != null) {
+							return javadoc;
+						}
+						// otherwise, maybe a supertype has it
+					}
+				} catch (Exception e) {
+					LOGGER.log(Level.SEVERE,
+							"Error while getting method signature of '" + method.getElementName() + "'.", e);
+				}
+			}
+		}
+		
+		// 3) Check the superclasses for the member
+
+		// Collect type extensions
+		List<String> extendedTypes = null;
+		if (type.isInterface()) {
+			IType[] interfaces = findImplementedInterfaces(type, monitor);
+			if (interfaces != null && interfaces.length > 0) {
+				extendedTypes = Stream.of(interfaces) //
+						.map(interfaceType -> interfaceType.getFullyQualifiedName()) //
+						.collect(Collectors.toList());
+			}
+		} else {
+			// ex : String implements CharSequence, ....
+			ITypeHierarchy typeHierarchy = type.newSupertypeHierarchy(monitor);
+			IType[] allSuperTypes = typeHierarchy.getSupertypes(type);
+			extendedTypes = Stream.of(allSuperTypes) //
+					.map(superType -> superType.getFullyQualifiedName()) //
+					.collect(Collectors.toList());
+		}
+
+		if (extendedTypes != null) {

```

Comment:
Instead of computing a list of String extendedTypes  computed from IType[], it should be perhaps better to work only with IType to avoid using getTypeFromParams although the IType is already known.

The signature could look like this:

`public String getJavadoc(Type type, QuteJavadocParams params, IJDTUtils utils, IProgressMonitor monitor, Set<String> visited) throws JavaModelException {`